eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/RSMergeWriter.java

```diff
@@ -194,49 +235,15 @@ private void insertRSRecordForMergeDestination(ClusteredVariantEntity mergeDesti
         ClusteredVariantEntity targetRS = mergeCandidates.stream().filter(rs -> rs.getAccession()
                                                                                   .equals(targetRSAccession))
                                                          .findFirst().get();
-        // If the target RS has already been merged into another RS due to a previous merge, get that RS
-        targetRS = getFinalMergeDestination(targetRS);
         List<ClusteredVariantEntity> mergees = mergeCandidates.stream()
                                                               .filter(rs -> !rs.getAccession()
                                                                                .equals(targetRSAccession))
                                                               .collect(Collectors.toList());
         return new ImmutablePair<>(targetRS, mergees);
     }
 
-    private ClusteredVariantEntity getFinalMergeDestination(ClusteredVariantEntity originalRS) {
-        Query queryToFindMergeDestination = query(
-                where(ACCESSION_ATTRIBUTE).is(originalRS.getAccession())).addCriteria(
-                where(EVENT_TYPE_ATTRIBUTE).is(EventType.MERGED.toString())).addCriteria(
-                where(INACTIVE_OBJECT_ATTRIBUTE + "." +
-                              REFERENCE_ASSEMBLY_FIELD_IN_CLUSTERED_VARIANT_COLLECTION)
-                        .is(this.assemblyAccession));
-        ClusteredVariantOperationEntity mergeOperationsWithRSAsMergeeInCVOE =
-                this.mongoTemplate.findOne(queryToFindMergeDestination, ClusteredVariantOperationEntity.class);
-        DbsnpClusteredVariantOperationEntity mergeOperationsWithRSAsMergeeInDbsnpCVOE =
-                this.mongoTemplate.findOne(queryToFindMergeDestination, DbsnpClusteredVariantOperationEntity.class);
-        if (Objects.nonNull(mergeOperationsWithRSAsMergeeInCVOE)) {
-            ClusteredVariantEntity mergee =
-                    getMergedClusteredVariantEntityFromOperation(mergeOperationsWithRSAsMergeeInCVOE);
-            return getFinalMergeDestination(mergee);
-        }
-        if (Objects.nonNull(mergeOperationsWithRSAsMergeeInDbsnpCVOE)) {
-            ClusteredVariantEntity mergee =
-                    getMergedClusteredVariantEntityFromOperation(mergeOperationsWithRSAsMergeeInDbsnpCVOE);
-            return getFinalMergeDestination(mergee);
-        }
-        return originalRS;
-    }
-
-    private ClusteredVariantEntity getMergedClusteredVariantEntityFromOperation(
-            EventDocument<IClusteredVariant, Long, ? extends ClusteredVariantInactiveEntity>
-                    clusteredVariantOperationEntity) {
-        ClusteredVariantInactiveEntity mergee = clusteredVariantOperationEntity.getInactiveObjects().get(0);
-        return new ClusteredVariantEntity(clusteredVariantOperationEntity.getMergedInto(),
-                                          mergee.getHashedMessage(), mergee.getModel(), mergee.getVersion());
-    }
```

Comment:
Deleted because we no longer need to look for final merge destination for a given RS since we [constantly update unprocessed merge candidates](https://docs.google.com/spreadsheets/d/1KQLVCUy-vqXKgkCDt2czX6kuMfsjfCc9uBsS19MZ6dY/edit#rangeid=1846669126) with the correct target RS based on preceding merges.